src/common/exif.cc

```diff
@@ -1740,18 +1740,14 @@ int dt_exif_read_blob(uint8_t **buf, const char *path, const int imgid, const in
     {
       exifData["Exif.Image.XResolution"] = Exiv2::Rational(resolution, 1);
       exifData["Exif.Image.YResolution"] = Exiv2::Rational(resolution, 1);
-      exifData["Exif.Image.ResolutionUnit"] = uint16_t(2); /* inches */
     }
     else
     {
-      static const char *keys[] = {
-        "Exif.Image.XResolution",
-        "Exif.Image.YResolution",
-        "Exif.Image.ResolutionUnit"
-      };
-      static const guint n_keys = G_N_ELEMENTS(keys);
-      dt_remove_exif_keys(exifData, keys, n_keys);
+      // mandatory tags for TIFF/EP and Exif, use Exif default
+      exifData["Exif.Image.XResolution"] = Exiv2::Rational(72, 1);
+      exifData["Exif.Image.YResolution"] = Exiv2::Rational(72, 1);
```

Comment:
As shown in https://github.com/darktable-org/darktable/blob/67c69966ebfd630479916dd36d7d42203f65b65e/data/darktableconfig.xml.in#L2570-L2572 the default for `darktable` is 300, min is 72 and max is 9600.

So i'd recommend using
```c
dt_confgen_get_int("metadata/resolution", DT_DEFAULT);
```

instead of `72`

Reply:
Cool, does that mean I can do away with the if block altogether, i.e. `dt_confgen_get_int()` is guaranteed to return a non-zero value?

Reply:
This is how `dt_conf_get_int` works:

```c
  const int min = dt_confgen_get_int(name, DT_MIN);
  const int max = dt_confgen_get_int(name, DT_MAX);
  const int val = dt_conf_get_int_fast(name);
  const int ret = CLAMP(val, min, max);
  return ret;
```

So result of `dt_conf_get_int` is guaranteed to be between MIN and MAX configured in `darktableconfig.xml.in`

for reading of actual value don't use `dt_confgen_get_int` as it gets ONLY defined defaults (min/max/def)

Reply:
Ah, so the existing code is incorrect anyway since it tests `resolution > 0` which is always true (min is 72)...

Reply:
Safety precauction from times before.

Reply:
To me it looks like dt_conf_get_int_fast() will give me the default value if it doesn't exist, so dt_conf_get_int() should already return the default in that case, not the min, which means the test and the other branch are completely unnecessary anyway?

Reply:
Pretty much I think. (Although having layers of checks "just in case" isn't too bad IMO)